karma-dashboard.config.cjs

```diff
@@ -161,6 +162,25 @@ module.exports = function (config) {
     // how many browsers should be started simultaneously
     concurrency: Infinity,
 
+    // Sharding.
+    parallelOptions: {
+      shardStrategy: typeof config.shardIndex !== 'undefined' && typeof config.executors !== 'undefined' ? 'custom' : 'round-robin',
+      // If we're using custom sharding, just spin up 1 browser,
+      // but do the splitting in the custom strategy below.
+      executors: typeof config.shardIndex !== 'undefined' ? 1 : undefined, // Default is cpu cores - 1.
+      shardIndex: typeof config.shardIndex !== 'undefined' ? Number(config.shardIndex) : undefined,
+      // Re-implements a round-robin strategy, but with a custom shardIndex.
+      // Need to use the Function constructor here so we have access to config.shardIndex,
+      // because karma-parallel serializes this function.
```

Comment:
I'm a little confused here. So we wouldn't be able to access the `shardIndex` from the config that's injected into the `customShardStrategy`?

This worked for me:
```js
    customShardStrategy: (config2) => {
        window.parallelDescribeCount = window.parallelDescribeCount || 0;
        window.parallelDescribeCount++;
        const shouldRunThisTest =
          window.parallelDescribeCount % Number(config2.executors) ===
          Number(config2.shardIndex);
        console.log(
          '[SHARDING DEBUG]',
          'Count:',
          window.parallelDescribeCount,
          'Executors:',
          Number(config2.executors),
          Number(config2.shardIndex),
          'Should run test?',
          shouldRunThisTest
        );
        return shouldRunThisTest;
      },```

Reply:
Hmm I initially had some issues with the one from config2 not having the correct values, so I wanted to access the one I passed via CLI. Will take another look.
Of course it would be great if it‘s not needed.

Reply:
Oh, we need access to the outside executors variable which is 4, whereas config2.executors is 1 in this PR. So I don‘t see how yours works

Reply:
I see what you're saying now. The outside `executors` and the inside `executors` are not the same value. We need to keep the `executors` in the parallel config at 1 while the outside one needs to be dynamic 👍

Reply:
Yes exactly!

I‘ll try to document this better.